sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientCodeSnippets.java

```diff
@@ -0,0 +1,109 @@
+
+import java.util.HashMap;
+import java.util.List;
+
+public class TableClientCodeSnippets {
+
+    public static void methods() {
+
+        //create a tableServiceClient
+        TableServiceClient tableServiceClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildClient();
+
+        //create TableClient
+        TableClient tableClient = new TableClientBuilder()
+            .connectionString("connectionString")
+            .tableName("OfficeSupplies")
+            .buildClient();
+
+
+        //create a table
+        try {
+            tableServiceClient.createTable("OfficeSupplies");
+        } catch (TableServiceErrorException e) {
+            //use azure core errors? based on
+            System.out.println("Create Table Unsuccessful. Error: " + e);
+        }
+
+        //delete  table
+        try {
+            tableServiceClient.deleteTable("OfficeSupplies");
+        } catch (TableServiceErrorException e) {
+            System.out.println("Delete Table Unsuccessful. Error: " + e);
+        }
+
+        //query tables
+        String selectString = "$selectString= TableName eq 'OfficeSupplies'";
+
+        try {
+            List<AzureTable> responseTables = tableServiceClient.queryTables(selectString);
+        } catch (HttpResponseException e) {
+            System.out.println("Table Query Unsuccessful. Error: " + e);
+        }
+
+
+        //insert entity
+        String tableName = "OfficeSupplies";
+        String row = "crayola markers";
+        String partitionKey = "markers";
+        HashMap<String, Object> tableEntityProperties = new HashMap<>();
+        TableEntity tableEntity = new TableEntity(tableName, row, partitionKey, tableEntityProperties);
+        try {
+            tableEntity = tableClient.insertEntity(tableEntity);
+        } catch (HttpResponseException e) {
+            System.out.println("Insert Entity Unsuccessful. Error: " + e);
+        }
+
+
+        //update entity
+        tableEntity.addProperty("Seller", "Crayola");
+        try {
+            tableClient.updateEntity(tableEntity);
+        } catch (HttpResponseException e) {
+            System.out.println("Update Entity Unsuccessful. Error: " + e);
+        }
+
+
+        //upsert entity (where it is an insert or replace)
+        tableEntity.addProperty("Price", "5");
+        try {
+            tableClient.insertOrReplaceEntity(tableEntity);
+        } catch (HttpResponseException e) {
+            System.out.println("Upsert Entity Unsuccessful. Error: " + e);
+        }
+
+        //upsert entity (where it is an insert or merge)
+        tableEntity.addProperty("Price", "5");
+        try {
+            tableClient.insertOrMergeEntity(tableEntity);
+        } catch (HttpResponseException e) {
+            System.out.println("Upsert Entity Unsuccessful. Error: " + e);
+        }
+
+
+        //delete entity
+        try {
+            tableClient.deleteEntity(tableEntity);
```

Comment:
Would it make sense to have a version of `deleteEntity` that just takes the `row` and `partitionKey` parameters? To avoid forcing the customer to get a `TableEntity` object just to delete it from the table (also if every API call costs the customer money, making a `delete` go from 2 to 1 REST calls is important).

Unlike the update/upsert operations, I don't *think* we need to provide an ETag in order to delete - I don't believe there is a potential concurrency problem there because we're not mutating properties on the row. If the service does require it then I guess there's nothing we can do here.

Reply:
Did we still want to resolve this comment?